go/test/apis/git_recognizer_test.go

```diff
@@ -17,61 +17,55 @@ import (
 	"github.com/redhat-developer/alizer/go/test"
 )
 
+const (
+	repoUrl    = "https://github.com/devfile-resources/alizer-test-resources.git"
+	repoCommit = "3f89d2f05fad6aeae6d386f5c7fa722124b8c017"
+)
+
 func TestExternalRepos(t *testing.T) {
-	// read git test file to retrieve all git repos and their expected properties
+	// read git test file to retrieve all test resources and their expected properties
 	jsonFile, err := ioutil.ReadFile("../git_test.json")
 	if err != nil {
 		t.Fatal("Unable to fetch git repositories file to run tests to")
 	}
-	var data test.GitTests
-	err = json.Unmarshal(jsonFile, &data)
+	var data []test.GitTestProperties
+	err = json.Unmarshal([]byte(jsonFile), &data)
 	if err != nil {
 		t.Fatal("Unable to fetch git repositories file to run tests to")
 	}
 
-	resChan := make(chan resultTest)
-	// loop over all repositories and verify expected results are correct
-	for repo, properties := range data {
-		checkoutAndTest(resChan, repo, properties)
+	// checkout to github repo
+	root, err := test.CheckoutCommit(repoUrl, repoCommit)
```

Comment:
wdyt about changing `test.CheckoutCommit` to include a dir arg. so instead of using `ioutil.TempDir`, we would validate that the dir passed in exists and then continue with the clone

this way, we could just pass in `t.TempDir()` and it will take care of the cleaning
https://pkg.go.dev/testing#B.TempDir